File: API_R_01Agosto2024/Kawã Sousa/q1_nomes.py

def main():
    ##Entrada
    nome = input("Qual o seu nome? ")
    anoNascimento = int(input("Em que ano você nasceu? "))
    idade = 2024 - anoNascimento

    numeroAtual = 1
    somaDivisores = 0
    numeroDivisores = 0
    maiorDivisorPrimo = 1
    
    print("Sua idade: ", idade)
    ##Processamento
    while numeroAtual <= idade:
        if idade % numeroAtual == 0:
            print(f"o número {numeroAtual} é divisor de {idade}")
            somaDivisores += numeroAtual
            numeroDivisores += 1
            
            ehPrimo = True
            for numero in range(numeroAtual):
                if numeroAtual % numero == 0:
                    ehPrimo = False
                    if ehPrimo:
                        if numero > maiorDivisorPrimo:
                            maiorDivisorPrimo = numero

        numeroAtual += 1

    mediaDivisores = somaDivisores / numeroDivisores if somaDivisores > 0 else 0
    ##Saída
    for numero in maiorDivisorPrimo:
        print(nome)
    

main()

File: API_R_01Agosto2024/Joao victor de sousa dos santos/API-R#01/q1_nomes.js
// q1_nomes

var input = require("readline-sync");

function main(){

    // entrada de dados
    var nome = input.question("\n> Informe seu nome: ");
    var ano_nascimento = input.questionInt("> Informe seu ano de nascimento: ");

    // processamento
    var idade = calcular_idade(ano_nascimento);
    console.log(`\n> Sua idade: ${idade}`)
    
    // exibindo divisores
    var maior_divisor_primo = null;
    var contador = 1;
    var somatorio_divisores = 0;
    var contador_divisores = 0;
    var media_divisores;
    
    console.log("> divisores:")
    while (contador < idade){
        var eh_divisor = idade % contador == 0;
        if (eh_divisor){
            console.log(contador);
            if (eh_primo(contador)){
                maior_divisor_primo = contador;
            }
            somatorio_divisores += contador;
            contador_divisores++;
        }
        contador++;
    }

    media_divisores = somatorio_divisores / contador_divisores;

    // se existir um maior divisor primo
    if (maior_divisor_primo != null){
        console.log(`\n> maior divisor primo: ${maior_divisor_primo}\n> imprindo seu nome ${maior_divisor_primo} vezes: `)
        var contador = 1;
        var d_vezes = maior_divisor_primo;
        while (contador <= d_vezes){
            console.log(nome)
            contador++;
        }
        return;
    } 
    
    console.log(`\n> media dos divisores: ${media_divisores.toFixed(2)}\n> imprindo seu nome ${media_divisores} vezes: `)
    // imprimindo o nome V_vezes(média dos divisores)
    var contador = 1;
    var v_vezes = media_divisores;
    while(contador <= v_vezes){
        console.log(nome);
        contador++;
    }

}

// verifica se um valor é primo
function eh_primo(valor){
    if (valor == 1){
        return false
    } else if (valor ==2 || valor==3){
        return true
    } else if ((valor % 2)==0 || (valor % 3)==0){
        return false
    } else{
        return true
    }
}

// calcula a idade
function calcular_idade(ano_nascimento){
    const ano_atual = 2024;
    return ano_atual - ano_nascimento;
}

main();